lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -845,7 +845,7 @@ public function getSingleResult($hydrationMode = null)
      *
      * Alias for getSingleResult(HYDRATE_SINGLE_SCALAR).
      *
-     * @return mixed The scalar result.
+     * @return string|null The scalar result.
```

Comment:
```suggestion
     * @return int|string|null The scalar result.
```

It's not perfect, but it fits to the current situation for 2.7.

Reply:
It doesn't fit.

I just tried
@ORM\Column(type="boolean") is return as boolean
@ORM\Column(type="datetime") is return as DateTime object
@ORM\Column(type="array") is return as an Array.

We certainly can get a float too.

I never used binary/blob but this is certainly convert to resource
Same for date_immutable and DateTimeImmutbale or dateinterval and DateInterval

And there is an object type, I never used it but maybe it return an object ?

Which lead to
```
boolean|int|float|string|\DateTime|\DateTime|\DateInterval|resource|array|object|null
```
Which is mixed, isn't it ?

The only type I don't know is resource and object, since I never used them.

Source:
https://www.doctrine-project.org/projects/doctrine-dbal/en/2.10/reference/types.html

Reply:
Oh, right :-D